hazelcast/src/test/java/com/hazelcast/spi/impl/sequence/FailFastCallIdSequenceTest.java

```diff
@@ -0,0 +1,148 @@
+
+        // backup-aware operation
+        testNext(new DummyBackupAwareOperation());
+
+        // urgent operation
+        testNext(new DummyPriorityOperation());
+    }
+
+    private void testNext(Operation operation) {
+        long oldSequence = sequence.getLastCallId();
+        long result = nextCallId(sequence, operation.isUrgent());
+        assertEquals(oldSequence + 1, result);
+        assertEquals(oldSequence + 1, sequence.getLastCallId());
+    }
+
+    @Test (expected = HazelcastOverloadException.class)
+    public void next_whenNoCapacity_thenThrowException() throws InterruptedException {
+        sequence = new FailFastCallIdSequence(1);
+        final long oldLastCallId = sequence.getLastCallId();
+
+        final CountDownLatch nextCalledLatch = new CountDownLatch(1);
+        spawn(new Runnable() {
+            @Override
+            public void run() {
+                DummyBackupAwareOperation op = new DummyBackupAwareOperation();
+                long callId = nextCallId(sequence, op.isUrgent());
+                setCallId(op, callId);
+                nextCalledLatch.countDown();
+                sleepSeconds(3);
+                sequence.complete();
+            }
+        });
+
+        nextCalledLatch.await();
+
+        nextCallId(sequence, false);
+    }
+
+    @Test
+    public void next_whenNoCapacity_thenBlockTillTimeout() {
+        sequence = new FailFastCallIdSequence(1);
+
+        // first invocation consumes the available call ID
+        nextCallId(sequence, false);
+
+        long oldLastCallId = sequence.getLastCallId();
+        try {
+            sequence.next();
+            fail();
+        } catch (HazelcastOverloadException e) {
+            // expected
+        }
+
+        assertEquals(oldLastCallId, sequence.getLastCallId());
+    }
+
+    @Test
+    public void when_overCapacityButPriorityItem_then_noBackpressure() {
+        final FailFastCallIdSequence sequence = new FailFastCallIdSequence(1);
+
+        // occupy the single call ID slot
+        nextCallId(sequence, true);
+
+        long oldLastCallId = sequence.getLastCallId();
+
+        long result = nextCallId(sequence, true);
+        assertEquals(oldLastCallId + 1, result);
+        assertEquals(oldLastCallId + 1, sequence.getLastCallId());
+    }
+
+    @Test
+    public void whenComplete_thenTailIncrements() {
+        nextCallId(sequence, false);
+
+        long oldSequence = sequence.getLastCallId();
+        long oldTail = sequence.getTail();
+        sequence.complete();
+
+        assertEquals(oldSequence, sequence.getLastCallId());
+        assertEquals(oldTail + 1, sequence.getTail());
+    }
+
+    @Test(expected = AssertionError.class)
+    @RequireAssertEnabled
+    public void complete_whenNoMatchingNext() {
+        nextCallId(sequence, false);
+        sequence.complete();
+        sequence.complete();
+    }
+
+    static long nextCallId(CallIdSequence seq, boolean isUrgent) {
```

Comment:
I would get rid of this method. It doesn't improve readability at all.

Reply:
OK, changed.